lib/redcord/migration/index.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+# typed: strict
+
+module Redcord::Migration::Index
+  extend T::Sig
+
+  sig { params(model: T.class_of(Redcord::Base), index_name: Symbol).void }
+  def remove_index(model, index_name)
+    if model.redis.sismember("#{model.model_key}:index_attrs", index_name)
+      _remove_inex_from_attr_set(
+        model: model,
+        attr_set_name: 'index_attrs',
+        index_name: index_name,
+      )
+
+      model.redis.scan_each(match: "#{model.model_key}:#{index_name}:*") { |key| _del_set(model, key) }
+    elsif model.redis.sismember("#{model.model_key}:range_index_attrs", index_name)
+      _remove_inex_from_attr_set(
+        model: model,
+        attr_set_name: 'range_index_attrs',
+        index_name: index_name,
+      )
+
+      model.redis.scan_each(match: "#{model.model_key}:#{index_name}:") { |key| _del_set(model, key) }
```

Comment:
I think you can use `sscan_each` here to directly iterate over the set (since we know what the key is for the nil range index set)

Reply:
well we still need to loop trough all the sets tho. given a set, we still need to call spop in sscan_each, so not entirely necessary i think.

Reply:
We know exactly what the key of the set is in this case, so we shouldn't have to loop through all the sets right? I think scan with match will still result in more iterations since it has to scan through all keys

https://redis.io/commands/scan#the-match-option

Reply:
oooh right there's only one set here lol